### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by udan11**
This PR contains various performance improvements and uploads-related bug fixes. See each individual commit.

### CODE REVIEW

File: assets/javascripts/lib/uploads.js [hunk 1]
```diff
@@ -47,3 +54,37 @@ export function generateUploadKey() {
...
+    };
+    req.onerror = reject;
+    req.send(null);
+  });
+
+  return Promise.all([keyPromise, downloadPromise]).then(([key, download]) => {
+    const iv = download.buffer.slice(0, 12);
+    const content = download.buffer.slice(12);
+
+    return new Promise((resolve, reject) => {
```
Comment (eviltrout) [inline assets/javascripts/lib/uploads.js:81]: I'm curious about the need to do `new Promise` here when `window.crypto.subtle.decrypt` seems to return a promise? Could you not return that?

File: assets/javascripts/lib/uploads.js [hunk 2]
```diff
@@ -47,3 +54,37 @@ export function generateUploadKey() {
...
+    req.onload = function () {
+      let filename = req.getResponseHeader("Content-Disposition");
+      if (filename) {
+        // Requires Access-Control-Expose-Headers: Content-Disposition.
+        filename = filename.match(/filename="(.*?)"/)[1];
+      }
+      resolve({ buffer: req.response, filename });
+    };
+    req.onerror = reject;
+    req.send(null);
```
Comment (eviltrout) [inline assets/javascripts/lib/uploads.js:74]:
```suggestion
    req.send();
```
This should be the same as `null` right?

**Review by eviltrout — APPROVED** [review]
I love this commit. It really brings us up to the newer Ember standards.

File: assets/javascripts/lib/uploads.js (hunk 1 above)
Comment (udan11) [reply assets/javascripts/lib/uploads.js:81]: I can, but the tests might not work anymore. The runloop is aware of Ember Promises, but is not of the native ones:

https://discuss.emberjs.com/t/readers-questions-why-does-ember-still-use-rsvp/14736

File: assets/javascripts/lib/uploads.js (hunk 2 above)
Comment (udan11) [reply assets/javascripts/lib/uploads.js]: Yes, you are right. MDN says the same.

**Pull request merged**